### PR TITLE
Ignore status code when making head request to visualstudio.com

### PR DESCRIPTION
**Pull request by chrisrpatterson**
Account level urls, https://<accountname>.visualstudio.com return a 404 when performing a HEAD request as there is no controller at that level to answer.  However, the HEAD request still returns the necessary headers expected by this module.  This change simply ignores the status code of the response and checks for the x-vss-resourcetenant header before throwing an error.

### CODE REVIEW

**Comment by chrisrpatterson** [comment]
Updating with new changes

**Comment by whoisj** [comment]
:+1: LGTM

File: src/deviceflow.ts [hunk 1]
```diff
@@ -232,24 +232,41 @@ export class DeviceFlowAuthenticator {
...
+            const tenantIds: string[] = resourceTenant.split(',');
+            if (tenantIds.length > 1) {
+                for (const id of tenantIds) {
+                    if (id !== "00000000-0000-0000-0000-000000000000") {
+                        tenantId = id;
+                    }
                 }
-            });
+            } else {
+                tenantId = tenantIds[0];
```
[the remark this answers is not in the document]
Comment (chrisrpatterson) [reply src/deviceflow.ts]: If no tenant id is returned it will throw

File: src/deviceflow.ts [hunk 2]
```diff
@@ -232,24 +232,41 @@ export class DeviceFlowAuthenticator {
...
+        } catch (error) {
+            resourceTenant = error.response['x-vss-resourcetenant'];
+        }
+
+        if (resourceTenant) {
+            const tenantIds: string[] = resourceTenant.split(',');
+            if (tenantIds.length > 1) {
+                for (const id of tenantIds) {
+                    if (id !== "00000000-0000-0000-0000-000000000000") {
+                        tenantId = id;
```
[the remark this answers is not in the document]
Comment (chrisrpatterson) [reply src/deviceflow.ts]: yep should do that

**Pull request merged**